hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTaskWithFSO.java

```diff
@@ -175,10 +121,102 @@ public Pair<Integer, Boolean> processWithFSO(OMUpdateEventBatch events,
     if (!flushAndCommitNSToDB(nsSummaryMap)) {
       return new ImmutablePair<>(seekPos, false);
     }
+    // Delete hard deleted directories from NSSummary table
+    deleteNSSummariesFromDB(objectIdsToBeDeleted);
```

Comment:
objectsToBedeleted can be part of flushAndCommitNSToDB(), so that flush can be done togethere.

same list can be passed for both call to ensure its fully deleted.